### ISSUE TITLE
Wrong option selected after hitting "Enter" in the country field or any other "Combobox" based component

### ISSUE THREAD

**Issue by tarhi-saad** [woocommerce/woocommerce-gutenberg-products-block#5872]
## Describe the bug

We have noticed that after copy-pasting a country name or typing it in the `Country/Region` field in the `Checkout Block`, hitting the "Enter" key from the keyboard will select the previously selected country instead.

This bug can be seen on all `Combobox` based components (e.g. `CountryInput`, `StateInput`)

## To reproduce

1. Create a test page and add the Checkout Block
2. Add a product to the cart and proceed to the checkout block
3. Scroll down to the `Billing address` section and select a country
4. Then type or copy-paste the name of a new country, then hit "Enter" to select. You'll notice that the previous country gets selected each time

## Expected behavior

While typing a country name or copy-pasting it, the first match should be selected at all times. This way, each time we hit "enter" the expected option will be chosen.

We can view the working version of this behavior on the default Checkout page under the "Country / Region" field.

### No preselected option while typing ❌

<img src="https://user-images.githubusercontent.com/14235870/154381718-0aa643fc-a10b-4796-b6a3-9b91beb98944.png"  alt="1" width = "360" >

### No preselected option when the exact option name is typed or copy-pasted ❌

<img src="https://user-images.githubusercontent.com/14235870/154382002-7c40a76c-fb91-4568-95f5-d057ceb953e1.png"  alt="1" width = "360" >

### The first match is preselected on the same field on the default checkout page ✅

<img src="https://user-images.githubusercontent.com/14235870/154383140-8007eca6-f937-4491-b7c0-8fbda61be3aa.png"  alt="1" width = "360" >

### WordPress (please complete the following information):

-   Core version: 5.9
-   Gutenberg version (if installed): 12.5.4
-   WooCommerce version: 6.2.0
-   WooCommerce Blocks version: 7.0.0
-   Site language: English
-   Any other plugins installed: WooCommerce Cart Add-Ons, WooCommerce Payments

### Desktop (please complete the following information):

-   OS: macOS
-   Browser chrome
-   Version 98.0.4758.102
